src/main/java/com/joejoe2/chat/controller/PrivateChannelController.java

```diff
@@ -247,4 +247,39 @@ public Object subscribe(@ParameterObject @Valid SubscribeRequest request)
       throws UserDoesNotExist {
     return channelService.subscribe(AuthUtil.currentUserDetail().getId());
   }
+
+  @Operation(summary = "set block state of private channel")
+  @AuthenticatedApi
+  @SecurityRequirement(name = "jwt")
+  @ApiResponses(
+      value = {
+        @ApiResponse(
+            responseCode = "403",
+            description = "you are not the member in target channel",
+            content =
+                @Content(
+                    mediaType = "application/json",
+                    schema = @Schema(implementation = ErrorMessageResponse.class))),
+        @ApiResponse(
+            responseCode = "404",
+            description = "target channel is not exist",
+            content =
+                @Content(
+                    mediaType = "application/json",
+                    schema = @Schema(implementation = ErrorMessageResponse.class))),
+        @ApiResponse(responseCode = "200", content = @Content),
+      })
+  @RequestMapping(path = "/blockage", method = RequestMethod.POST)
+  public ResponseEntity<Object> setBlockage(@Valid @RequestBody ChannelBlockRequest request)
+      throws UserDoesNotExist {
```

Comment:
Perhaps we can catch this exception as well.

Also, I don't think it should be checked. I'd prefer to make this one unchecked RuntimeException. Both as per semantics and use case.

Also point to consider is an introduction of proper ExceptionAdvice which would make all the exceptions neat in response having a unified response body which would hide stacktrace and all the sensitive information.

In general, it's advice to perform exception handling in the service layer instead.

Reply:
This exception wil never occur at controller level. But may occur when someone just call the service directly. So I just throw out and let the global exception handler of spring to proccess that.

Reply:
Sounds reasonable. Thanks for sorting it out!

Reply:
If we catch this exception within the service clas it's still gonna be picked up by the global spring exception handler.

In general it's advised having all the exception handling within the service layer.